itenium-socks/src/app/socks/sock.component.html

```diff
@@ -32,14 +32,20 @@ <h6><span>€</span><span>{{ sock.price }}</span></h6>
       </div>
       <div class="col-sm-6 col-md-6 col-lg-6">
         <h1 style="margin-top: 25px;">{{ sock.name }}</h1>
+        @if (sock.inventory == 0) {
+        <span class="soldOutBadge">SOLD OUT!! 😭</span>
+       }
+       @else {
         {{ sock.inventory }} in stock. FREE Shipping!
+       }
+        
         <br>
         Color: {{ sock.variant | titlecase }}
 
         <br>
         <br>
 
-        <button type="button" class="buy-socks" (click)="buy()">
+        <button id="buyBtn" *ngIf="sock.inventory > 0" type="button" class="buy-socks" (click)="buy()">
```

Comment:
Avoid `<HTMLInputElement> document.getElementById("buyBtn");` --> it's mixing together pure JavaScript & Angular.

In this case you could use:

```
 <button [disabled]="buttonDisabled" *ngIf="sock.inventory > 0" type="button" class="buy-socks" (click)="buy()">
```


If you really need to manipulate a DOM object manually (this should be rare)

```
constructor(elementRef: ElementRef) {}
```

See: https://angular.dev/guide/components/dom-apis#